### PR TITLE
[Dessert] - Copilot

### PR DESCRIPTION
**Pull request by jz263**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# 告别面向google编程，copilot测试版本体验
+
+最近OpenAI和Github联合发布了一个名为copilot代码生成工具。这个工具基于GPT-3自然语言处理模型开发，并使用github上的数十亿行代码作为训练数据，实现在写代码时自动提供代码建议。
```
Comment (jhqian0909) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:3]: 的功能

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## copilot背后的模型GTP-3
+
+大家对GPT-3并不陌生 - 或许是2020年最火的自然语言处理模型，训练成本接近500万元，模型参数超过1750亿个。他是一个自回归模型，通俗理解就是输入一段文本的前一部分，预测接下来的文本的概率分布。GTP-3 自问世以来就被开发成几十个大大小小的不同应用，例如进行一些创作性写作（诗歌，小说等），编写前端代码（debuild.co）等等。
+
+鲁迅曾经说过，XXX是最好的语言，既然GTP-3在语言方面如此有天赋，那么在写代码方面应该也会有所建树吧。果然CodeX - 基于GTP-3微调后的编程语言版本也在不久后问世，而copilot正是CodeX的第一个商业化应用。
+
+## copilot试用体验
+
+copilot官网中(copilot.github.com)列举了copilot的四个使用方向：
```
Comment (jhqian0909) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:17]: 可以用Link的方式[]()

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# 告别面向google编程，copilot测试版本体验
+
+最近OpenAI和Github联合发布了一个名为copilot代码生成工具。这个工具基于GPT-3自然语言处理模型开发，并使用github上的数十亿行代码作为训练数据，实现在写代码时自动提供代码建议。
+
+目前该产品已经在vscode extenison中上线。作为长期面向google编程的小白，当然是偷懒等不了明天，早早在官网排队等待试用名额，幸运地在一两周前获得了试用机会。
+
+一周的体验下来，我认为copilot的功能相较于其他代码补齐工具而言更加强大，更加灵活，甚至有时让我有种仿佛copilot真的能读懂我的想法和代码，并且将其实现的感觉。显著提高了写码效率。
+
+## copilot背后的模型GTP-3
```
Comment (jhqian0909) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:9]: 小标题可以加1 1.1这样

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-7-20/1626788592731-copilot%20repetitive%20code.PNG" width="600" />
+  <figcaption>填充重复性代码</figcaption>
+</figure>
+
+结果是非常Amazing啊，可以看出copilot不仅提供了符合范式的函数名，代码以及符合项目代码的参数调用，并且对函数内容也有相对应的改动(request.post中传入的files参数由image变成了url）。说明copilot的确将本地文件中的代码作为了模型的输入值，并且很好地“理解”了代码中的重要特征，如格式，规范，结构等等。
+## 直观感受
+
+总体而言，虽然copilot生成的代码在多数情况下还需要检查和调整，但是他革命性的代码生成效果还是把初步接触copilot的我惊艳到了，有时就算程序已经写好，我还是会特意按下Ctrl+Enter查看copilot提供的不同思路和方法。
+
+最后夹带一点点个人观点，有人质疑copilot成熟以后或许会抢走部分程序员的饭碗，或者让程序员自主思考的能力逐渐退化，作为一个入门码农来说我目前觉得并不需要担心。Copilot给我的感觉像是一个专门提供给程序员的google，或是一套整理齐全，索引完备的代码答案库，在面对多数人遇到过的类似问题的时候，可以提高编程效率，减少面向google编程时搜索合适答案的时间；而另一方面，毕竟GTP-3本质上是一个自回归模型，我们无法依赖其提供给我们更加创新的算法或代码方案，当大多数程序员从重复且枯燥的代码中解放出来的时候，或许会就有更多的人类智慧被利用在创新性的工作中，从而加速行业的进步。
```
Comment (jhqian0909) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:49]: 感觉最后一段有点多。可以适当的换换行

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 2 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改成link

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 1 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 3 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 4 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# 告别面向google编程，copilot测试版本体验
+
+最近OpenAI和Github联合发布了一个名为copilot代码生成工具。这个工具基于GPT-3自然语言处理模型开发，并使用github上的数十亿行代码作为训练数据，实现在写代码时自动提供代码建议的功能。
```
Comment (BulletTech2021) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:3]: GitHub, please correct the typing

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+目前该产品已经在vscode extenison中上线。作为长期面向google编程的小白，当然是偷懒等不了明天，早早在官网排队等待试用名额，幸运地在一两周前获得了试用机会。
+
+一周的体验下来，我认为copilot的功能相较于其他代码补齐工具而言更加强大，更加灵活，甚至有时让我有种仿佛copilot真的能读懂我的想法和代码，并且将其实现的感觉。显著提高了写码效率。
+
+## 1. copilot背后的模型GTP-3
```
Comment (BulletTech2021) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:9]: 2级标题数字后不用加点

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+我用以下代码进行了一个简单测试，在DeepaiImageApi模块中，我想要通过两个不同函数分别实现将图片数据和图片url传给DeepAI提供的API接口。在我定义了第一个函数方法之后，来看看copilot是否能根据我的注释直接生成我想要的代码吧:
+
+<figure>
+  <img src="https://cdn.jsdelivr.net/gh/BulletTech2021/Pics/2021-7-20/1626788592731-copilot%20repetitive%20code.PNG" width="600" />
+  <figcaption>填充重复性代码</figcaption>
+</figure>
+
+结果是非常Amazing啊，可以看出copilot不仅提供了符合范式的函数名，代码以及符合项目代码的参数调用，并且对函数内容也有相对应的改动(request.post中传入的files参数由image变成了url）。说明copilot的确将本地文件中的代码作为了模型的输入值，并且很好地“理解”了代码中的重要特征，如格式，规范，结构等等。
+## 3. 直观感受
```
Comment (BulletTech2021) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:45]: 不同类型的元素要有空行，不然格式会乱

File: docs/Dessert/Apps/2021-07-19-Copilot.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+总体而言，虽然copilot生成的代码在多数情况下还需要检查和调整，但是他革命性的代码生成效果还是把初步接触copilot的我惊艳到了，有时就算程序已经写好，我还是会特意按下Ctrl+Enter查看copilot提供的不同思路和方法。
+
+最后夹带一点点个人观点，有人质疑copilot成熟以后或许会抢走部分程序员的饭碗，或者让程序员自主思考的能力逐渐退化，作为一个入门码农来说我目前觉得并不需要担心。Copilot给我的感觉像是一个专门提供给程序员的google，或是一套整理齐全，索引完备的代码答案库，在面对多数人遇到过的类似问题的时候，可以提高编程效率，减少面向google编程时搜索合适答案的时间。
```
Comment (BulletTech2021) [inline docs/Dessert/Apps/2021-07-19-Copilot.md:49]: Google，第一个字母要大写

**Review by BulletTech2021 — APPROVED** [review]
纠正了一些拼写问题。暂时没有看视频

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 5 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 6 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 7 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

File: docs/Dessert/Apps/2021-07-19-Copilot.md (hunk 8 above)
Comment (jz263) [reply docs/Dessert/Apps/2021-07-19-Copilot.md]: 已修改

**Pull request merged**